src/Servers/Kestrel/Core/src/Internal/Http2/FlowControl/InputFlowControl.cs

```diff
@@ -26,6 +26,11 @@ public InputFlowControl(uint initialWindowSize, uint minWindowSizeIncrement)
 
         public bool IsAvailabilityLow => _flow.Available < _minWindowSizeIncrement;
 
+        public void Reset()
+        {
+            _flow = new FlowControl((uint)_initialWindowSize);
```

Comment:
Not sure this is worth pooling. If we do pool, _pendingUpdateSize and _windowUpdatesDisabled should be reset.

Reply:
It's a struct yo

Reply:
I'm very aware. I was referring to InputFlowControl.cs.

Reply:
Ah ok. Now resetting those variables.